### DISCUSSION TITLE
Add Papago/Naver Translation option for Asiatic languages

### DISCUSSION THREAD

**Discussion by thealexmay** [SubtitleEdit/subtitleedit#7676] [Ideas]
In my opinion, Papago (from Naver) is one of the best online translators for Asiatic languages, especially Korean, but they have: Japanese, Korean, Chinese (Simplified & Traditional), Vietnamese, Thai, Indonesian, along with Russian, Spanish, Portugese, French, and German. Would love to see Papago added as an option for the Auto-Translate feature. [Here is a link](https://guide.ncloud-docs.com/docs/en/papagotranslation-api) to their API documentation.

**Comment by niksedk** [comment]
I've trouble signing up... if you could email me Client ID and the Client Secret, I could give it a try.

**Comment by thealexmay** [comment]
Did you have an issue with the phone verification? I'm in US and have had the same issue. I was hoping the cloud side of things would not have the same issue as their main site. I'll see if I can find a way around it.

**Comment by niksedk** [comment]
I'm in Denmark and that does not seem to be supported. 
Let me know if you find a way around it.

**Comment by thealexmay** [comment]
SO, I finally was able to get the phone verification to work, but the easy log-in feature is only for accounts verified with South Korean phone numbers. I attempted to register on the cloud site the traditional way, but it asks for business information including a business certificate for tax purposes, which obviously I don't have. Not even sure if you have that / are interested in a business account with Naver, but we could coordinate on creating an account for you to use, were that the case.

I was able to do an easy log-in using a friend's account from South Korea but it won't give me much details on anything without adding a payment method, and I don't want to compromise my friend's account in any sort of financial way. lol

It seems that the Naver API is not tremendously public-facing, in that it's not really public-facing at all. In terms of adding this to SubtitleEdit, that's not something you would usually pay for, correct? The cost is transferred to the user who needs an API key, am I understanding this all correctly? There's a possibility I could have a Korean friend essentially create a "burner" Naver account to use that I could just provide you the login info for, but all Naver accounts have to be verified with a phone number, so I just want to make sure I'm not teeing my friends up for a large bill.

Beyond all of that, seems like I'll be stuck using auto-translate via copy-paste with Papago, which is time consuming but better than nothing.

Editing to add here: I did a bit more digging and noticed an API inquiries link at the bottom of the Papago translator page that goes to a different place. They have a system to inquire about API usage and also business partnerships. And the API inquiry form has an option for Denmark phone numbers. Here are those links, in case useful to you:

- [Papago API page](https://papago.naver.com/business)
- [Naver Service Inquiry](https://www.ncloud.com/support/question/service?parentCode=O2O_AI&code=O2O_AI_005) (for "Type of Inquiry" at the top, you can search "Papago" and the option would be "AI Services - Papago Translation"
- [Naver Partnership Proposal Inquiry](https://www.navercloudcorp.com/en/ncp/proposalGuide)

**Comment by niksedk** [comment]
I've tried to implement it here (not tested though): https://github.com/SubtitleEdit/subtitleedit/releases/download/4.0.2/SubtitleEditBeta.zip

These need setting in Settings.xml 
````xml
    <AutoTranslatePapagoApiKeyId>...</AutoTranslatePapagoApiKeyId>
    <AutoTranslatePapagoApiKey>...</AutoTranslatePapagoApiKey>
````

**Comment by thealexmay** [comment]
Neat. I'll see if I can get it to work, then!
